### PR TITLE
feat/Deploy for PRs

### PR DESCRIPTION
**Pull request by markcsims**
https://asaphub.atlassian.net/browse/CRN-571

- [x] yarn sls deploy
- [x] e2e branch
- [x] verify branch
- [x] verify schema
- [x] define pipeline dependancies
- [x] Teardown sls 
- [x] Teardown environments left behind
- [x] Test Calendar
- [x] Test Squidex Data
- [x] Check env with no gitlab
- [x] Remove AWS_ECR_PASSWORD secret

### CODE REVIEW

File: apps/asap-cli/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
+    "composite": true,
```
Comment (peterstarling) [inline apps/asap-cli/tsconfig.json:4]: why do we need it?

File: apps/asap-server/test/handlers/webhooks/fetch-by-code/fetch-by-code.test.ts [hunk 2]
```diff
@@ -197,6 +197,14 @@ describe('Fetch-user-by-code handler', () => {
       });
 
       test('should ask for a key with expiration which is 601 minutes (10 hours 1 minute) ahead of now', async () => {
+        const dateReference = '2021-07-06T09:21:23.000Z';
```
Comment (peterstarling) [inline apps/asap-server/test/handlers/webhooks/fetch-by-code/fetch-by-code.test.ts:200]: strange that this is part of this PR no?

File: .github/workflows/build-pipeline.yml [hunk 3]
```diff
@@ -347,41 +449,176 @@ jobs:
...
     steps:
       - name: Checkout
-        uses: actions/checkout@v2
-      - uses: ./.github/actions/slack/
-        with:
-          message: 'Build Failed on GitHub: ${{ needs.setup.outputs.pr-number || github.head_ref }}'
-          webhook: ${{ secrets.SLACK_WEBHOOK }}
-          status: failure
+        uses: actions/checkout@v3
+      # - uses: ./.github/actions/slack/
```
Comment (peterstarling) [inline .github/workflows/build-pipeline.yml:593]: pretty sure we want this to happen no?

**Review by peterstarling — DISMISSED** [review]

File: apps/asap-server/test/handlers/webhooks/fetch-by-code/fetch-by-code.test.ts (hunk 2 above)
Comment (markcsims) [reply apps/asap-server/test/handlers/webhooks/fetch-by-code/fetch-by-code.test.ts:200]: Flaky test, decided to fix it.

File: .github/workflows/build-pipeline.yml (hunk 3 above)
Comment (markcsims) [reply .github/workflows/build-pipeline.yml:593]: I got asked to comment it out while we are developing, there was too much noise in the channel.

File: apps/asap-cli/tsconfig.json (hunk 1 above)
Comment (markcsims) [reply apps/asap-cli/tsconfig.json:4]: The "build-typecheck" step, generates .d.ts files for non-composite projects.

These types are then used in the "build sls package" and "build frontend".

asap-cli for example, doesn't need it's types build but it does need to be "typecheck-ed". Which is now done in the "typecheck" step.

File: .github/workflows/build-pipeline.yml (hunk 3 above)
Comment (markcsims) [reply .github/workflows/build-pipeline.yml]: I'll add it back, but we may need to re-think how and when we notify Slack.

Comment (peterstarling) [reply .github/workflows/build-pipeline.yml]: ahh i see, why is there so much noise though, i wonder? we do not notify in slack when the PR pipeline fails in gitlab, do we? in which case you probably want to remove it entirely, no?

Comment (markcsims) [reply .github/workflows/build-pipeline.yml]: Removed!

**Review by kmaid — APPROVED** [review]

**Pull request merged**